apps/src/code-studio/components/ShareAllowedDialog.jsx

```diff
@@ -172,6 +174,16 @@ class ShareAllowedDialog extends React.Component {
     this.props.onUnpublish(this.props.channelId);
   };
 
+  // Copy to clipboard.
+  copy = () => {
+    navigator.clipboard.writeText(this.props.shareUrl).then(
```

Comment:
we have a [copyToClipboard util function](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/util/copyToClipboard.js) that works slightly differently. i _think_ the main difference in implementations between that util function and yours here would just be browser compatibility? here's the [navigator.clipboard compatibility chart](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/clipboard#browser_compatibility) -- we may also want to double check the versions in that chart to make sure we don't break compatibility with browsers outside of IE.

if possible, it would be useful to have this utility be the same everywhere, so we may want to update it to use this implementation.

Reply:
I'm double-checking the browser compatibility, but in the meantime have updated and started using that shared function.  The function now takes optional `onSuccess` and `onFailure` handlers for when the async operation completes.

Reply:
As noted in the updated PR description, we now fall back to the legacy technique if the new one isn't available.